config/user-agents.json

```diff
@@ -0,0 +1,14 @@
+{
+"safari": {
+    "os": ["Macintosh; Intel Mac OS X 10_15_7", "Macintosh; Intel Mac OS X 10_15_6", 
+           "Macintosh; Intel Mac OS X 11_3", ""],
+    "versions": ["14.1.1", "14.1", "14.0.3"]
+    },
+
+"firefox": {
+    "os": ["Macintosh; Intel Mac OS X 10.15'", "Macintosh; Intel Mac OS X 10.14",
+            "Windows NT 10.0; Win64; x64", "X11; Ubuntu; Linux x86_64",
+            "X11; Linux x86_64", "Windows NT 10.0", "Macintosh; Intel Mac OS X 11.3"],
+    "versions": ["88.0 ", "89.0", "90.0", "78.0"]
+    }   
```

Comment:
```suggestion
  "os": {
    "macos_1": [
      "Macintosh; Intel Mac OS X 10_15_6", 
      "Macintosh; Intel Mac OS X 10_15_7",
      "Macintosh; Intel Mac OS X 11_3"
    ],
    "macos_2": [
      "Macintosh; Intel Mac OS X 10.14",
      "Macintosh; Intel Mac OS X 10.15",
    ],
    "windows_xp": [
      "Windows NT 5.1"
      "Windows NT 5.1; WOW64"
    ],
    "windows_vista": [
      "Windows NT 6.0"
      "Windows NT 6.0; WOW64"
    ],
    "windows_7": [
      "Windows NT 6.1"
      "Windows NT 6.1; WOW64"
      "Windows NT 6.1; Win64; x64"
    ],
    "windows_8": [
      "Windows NT 6.2; WOW64"
      "Windows NT 6.2; Win64; x64"
      "Windows NT 6.3; WOW64"
    ],
    "windows_10": [
      "Windows NT 10.0; WOW64"
      "Windows NT 10.0; Win64; x64"
    ],
    "linux": [
      "X11; Ubuntu; Linux x86_64",
      "X11; Linux x86_64",
    ],
  },
  "browsers": {
    "safari": [
      {
        "base": "Mozilla/5.0 (<os>) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/<version> Safari/605.1.15"
        "os": ["macos_1"],
        "versions": ["14.1.1", "14.1", "14.0.3"]
      }
    ],
    "firefox": [
      {
        "base": "Mozilla/5.0 (<os>; rv:<version>) Gecko/20100101 Firefox/<version>",
        "os": ["macos_2", "windows_vista", "windows_7", "windows_8", "windows_10", "linux64"],
        "versions": ["78.0", "88.0 ", "89.0", "90.0", "91.0"],
      }
    ],
    "opera": [
      {
        "base": "Opera/9.80 (<os>) <version>",
        "os": ["windows_xp", "windows_vista", "windows_7", "windows_8"],
        "versions": [
          "Presto/2.9.168 Version/11.50",
          "Presto/2.12.388 Version/12.14",
          "Presto/2.12.388 Version/12.16",
          "Presto/2.12.388 Version/12.16",
          "Presto/2.12.388 Version/12.17",
          "Presto/2.12.388 Version/12.17",
          "Presto/2.12.388 Version/12.17",
          "Presto/2.12.388 Version/12.18",
          "Presto/2.12.407 Version/12.51",
        ]
      },
      {
        "base": " Mozilla/5.0 (<os>) AppleWebKit/537.36 (KHTML, like Gecko) <version>",
        "os": ["windows_xp", "windows_vista", "windows_7", "windows_8", "windows_10"],
        "versions": [
          "Chrome/66.0.3359.170 Safari/537.36 OPR/53.0.2907.99",
          "Chrome/66.0.3359.170 Safari/537.36 OPR/53.0.2907.99",
          "Chrome/69.0.3497.100 Safari/537.36 OPR/56.0.3051.52",
          "Chrome/78.0.3904.108 Safari/537.36 OPR/65.0.3467.78",
          "Chrome/79.0.3945.130 Safari/537.36 OPR/66.0.3515.44",
          "Chrome/79.0.3945.130 Safari/537.36 OPR/66.0.3515.72",
          "Chrome/80.0.3987.132 Safari/537.36 OPR/67.0.3575.97",
        ]
      }
    ]
```

Reply:
The structure is a bit more complex, but allows more flexibility.

Reply:
Not sure if using agents from XP, Vista and 7 are a good idea. Those systems are far too dated. Also, I thought we're not using chromium-based browsers? Both Opera and Chrome should be removed.

Reply:
> Not sure if using agents from XP, Vista, 7 or 8 are a good idea. Those systems are far too dated.

According to whatismybrowser, those are still relatively frequent (of course, not at the level of more modern UAs). A weighting mechanism could be a good thing, yeah.

> Also, I thought we're not using chromium-based browsers? Both Opera and Chrome should be removed in that case.

Chromium should be fine, as it doesn't have the proprietary bits of chrome that Google could detect.

Reply:
> According to whatismybrowser, those are still relatively frequent

Source? I haven't dug deeper into [their site yet but it really seems like they just put `Very Common` for all user agents](https://developers.whatismybrowser.com/useragents/explore/hardware_type_specific/computer/8). Instead I think [the dynamic data on this site probably more trustworthy](https://techblog.willshouse.com/2012/01/03/most-common-user-agents) which puts Win7 and Win8 at around ~0.2% - 0.3%. 

>  Chromium should be fine, as it doesn't have the proprietary bits of chrome that Google could detect...

Ping  @unixfox. Wdyt?

Reply:
VR data likely isn't returned for some of the ancient user agents above.

Reply:
> VR data likely isn't returned for some of the ancient user agents above.

Well, then we should only keep UAs for NT>= 6.1 (win 7) (if possible, otherwise only keep NT >=10)